server/block/wood_sign.go

```diff
@@ -0,0 +1,119 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/block/model"
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/go-gl/mathgl/mgl64"
+	"math"
+	"strings"
+)
+
+// WoodSign is block that you can write text on.
+type WoodSign struct {
+	transparent
+	bass //might be wrong
+
+	// Wood is the type of wood of the door. This field must have one of the values found in the material
+	// package.
+	Wood WoodType
+	// Facing is the direction the sign is facing.
+	Facing cube.Direction
+	// Standing is whether the sign is on a block or against
+	Standing bool
+	// Text is an array with the text
+	Text string
+	// SignTextColor Color the sign is dyed witth
+	SignTextColor int32
+	// TextOwner XUID of the player who placed the sign
+	TextOwner string
+}
+
+// FlammabilityInfo ...
+func (s WoodSign) FlammabilityInfo() FlammabilityInfo {
+	return newFlammabilityInfo(0, 0, false)
+}
+
+// Model ...
+func (s WoodSign) Model() world.BlockModel {
+	return model.Empty{}
+}
+
+// BreakInfo ...
+func (s WoodSign) BreakInfo() BreakInfo {
+	return newBreakInfo(1, alwaysHarvestable, axeEffective, oneOf(s))
+}
+
+// CanDisplace ...
+func (s WoodSign) CanDisplace(l world.Liquid) bool {
+	_, water := l.(Water)
+	return water
+}
+
+func (s WoodSign) UseOnBlock(pos cube.Pos, face cube.Face, clickPos mgl64.Vec3, w *world.World, user item.User, ctx *item.UseContext) (used bool) {
+	pos = pos.Side(face)
+	if face != cube.FaceDown {
+
+		if face == cube.FaceUp {
+			s.Standing = true
+			s.Facing = cube.Direction(int16(math.Floor((((user.Yaw()+180)*16/360)+0.5)*100)/100) & 15)
```

Comment:
I don't think it's a good idea to merge this into the Facing field. The cube.Direction type can't have up to 16 values, it only knows of north, east, south and west. I think we need a separate type for this.